R/fixed_design_fh.R

```diff
@@ -0,0 +1,136 @@
+#  Copyright (c) 2023 Merck & Co., Inc., Rahway, NJ, USA and its affiliates.
+#  All rights reserved.
+#
+#  This file is part of the gsDesign2 program.
+#
+#  gsDesign2 is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#' Fixed design sample size
+#'
+#' Computes fixed design sample size for Fleming-Harrington method.
+#' Returns a tibble with a basic summary.
+#' @inheritParams gs_design_wlr
+#' @inheritParams gs_power_wlr
+#' @param power Power (`NULL` to compute power or strictly between 0
+#'   and `1 - alpha` otherwise).
+#' @param study_duration Study duration.
+#' @param rho test parameter in Fleming-Harrington method.
+#' @param gamma test parameter in Fleming-Harrington method.
+#' @return A table.
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' # example 1: given power and compute sample size
+#' x <- fixed_design_fh(
+#'   alpha = .025, power = .9,
+#'   enroll_rate = define_enroll_rate(duration = 18, rate = 1),
+#'   fail_rate = define_fail_rate(
+#'     duration = c(4, 100),
+#'     fail_rate = log(2) / 12,
+#'     hr = c(1, .6),
+#'     dropout_rate = .001
+#'   ),
+#'   study_duration = 36,
+#'   rho = 1, gamma = 1
+#' )
+#' x %>% summary()
+#' # example 2: given sample size and compute power
+#' x <- fixed_design_fh(
+#'   alpha = .025,
+#'   enroll_rate = define_enroll_rate(duration = 18, rate = 20),
+#'   fail_rate = define_fail_rate(
+#'     duration = c(4, 100),
+#'     fail_rate = log(2) / 12,
+#'     hr = c(1, .6),
+#'     dropout_rate = .001
+#'   ),
+#'   study_duration = 36,
+#'   rho = 1, gamma = 1
+#' )
+#' x %>% summary()
+fixed_design_fh <- function(alpha = 0.025,
```

Comment:
Could we call it `fixed_design_wlr_fh`?

Reply:
Is `fixed_design_wlr_fh` too long? We have 3 words there...
Let's keep it as `fixed_design_fh` for now. As long as we have clear documentation (https://merck.github.io/gsDesign2/articles/usage-fixed-design.html#introduction), I guess it will not be a big problem.